.github/workflows/on-pr-workflow.yml

```diff
@@ -2,19 +2,30 @@ name: Continuous Integration
 
 on:
   pull_request:
-    branches:
-        - master
 
   # Allows the manual execution of the workflow from the Action tab
   workflow_dispatch:
+   inputs:
+      enable_deployment:
+        description: 'Deploy to GitHub Pages? (true/false)'
+        required: false
+        default: false
+        type: boolean
 
-jobs:
-
-    lint-and-test:
-      uses: ./.github/workflows/lint-test-workflow.yml
-      with:
-        deploy-test-coverage: false
-
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+  security-events: write
 
+jobs:
+  lint-and-test:
+    name: Lint, test and deploy test coverage to Github Pages
+    uses: ./.github/workflows/lint-test-workflow.yml
+    with:
+      deploy: ${{ github.event.inputs.enable_deployment }}
 
-      
+  security-and-analysis:
+    name: Security and code analysis
+    uses: ./.github/workflows/security-analysis-workflow.yml
```

Comment:
missing end of file. Could maybe be automated.

Reply:
already automated, but only for source files (too much permissions problem with other files)